newton/_src/sim/builder.py

```diff
@@ -3178,6 +3183,66 @@ def add_shape_convex_hull(
             key=key,
         )
 
+    def add_site(
+        self,
+        body: int,
+        xform: Transform | None = None,
+        type: int = GeoType.SPHERE,
+        scale: Vec3 = (0.01, 0.01, 0.01),
+        key: str | None = None,
+        visible: bool = False,
+        custom_attributes: dict[str, Any] | None = None,
+    ) -> int:
+        """Adds a site (non-colliding reference point) to a body.
+
+        Sites are abstract markers that don't participate in physics simulation or collision detection.
+        They are useful for:
+        - Sensor attachment points (IMU, camera, etc.)
+        - Frame of reference definitions
+        - Debugging and visualization markers
+        - Spatial tendon attachment points (when exported to MuJoCo)
+
+        Args:
+            body (int): The index of the parent body this site belongs to. Use -1 for sites not attached to any specific body.
```

Comment:
Does -1 mean a static, world-space transform? If so, this could be a bit more concrete, e.g.
```suggestion
            body (int): The index of the parent body this site belongs to. Use -1 to define a site with a fixed transform in world frame.
```